pylintrc

```diff
@@ -0,0 +1,407 @@
+[MASTER]
+
+# Specify a configuration file.
+#rcfile=
+
+# Python code to execute, usually for sys.path manipulation such as
+# pygtk.require().
+#init-hook=
+
+# Profiled execution.
+profile=no
+
+# Add files or directories to the blacklist. They should be base names, not
+# paths.
+ignore=CVS
+
+# Pickle collected data for later comparisons.
+persistent=yes
+
+# List of plugins (as comma separated values of python modules names) to load,
+# usually to register additional checkers.
+load-plugins=
+
+# DEPRECATED
+include-ids=no
+
+# DEPRECATED
+symbols=no
+
+# Use multiple processes to speed up Pylint.
+jobs=1
+
+# Allow loading of arbitrary C extensions. Extensions are imported into the
+# active Python interpreter and may run arbitrary code.
+unsafe-load-any-extension=no
+
+# A comma-separated list of package or module names from where C extensions may
+# be loaded. Extensions are loading into the active Python interpreter and may
+# run arbitrary code
+extension-pkg-whitelist=
+
+
+[MESSAGES CONTROL]
+
+# Only show warnings with the listed confidence levels. Leave empty to show
+# all. Valid levels: HIGH, INFERENCE, INFERENCE_FAILURE, UNDEFINED
+confidence=
+
+# Enable the message, report, category or checker with the given id(s). You can
+# either give multiple identifier separated by comma (,) or put this option
+# multiple time. See also the "--disable" option for examples.
+#enable=
+
+# Disable the message, report, category or checker with the given id(s). You
+# can either give multiple identifiers separated by comma (,) or put this
+# option multiple times (only on the command line, not in the configuration
+# file where it should appear only once).You can also use "--disable=all" to
+# disable everything first and then reenable specific checks. For example, if
+# you want to run only the similarities checker, you can use "--disable=all
+# --enable=similarities". If you want to run only the classes checker, but have
+# no Warning level messages displayed, use"--disable=all --enable=classes
+# --disable=W"
+disable = 
+	locally-disabled,
+	too-few-public-methods,
+	bad-builtin,
+	star-args,
+	abstract-class-not-used,
+	abstract-class-little-used,
+	no-init,
+	fixme,
+	too-many-lines,
+	no-self-use,
+	too-many-ancestors,
+	too-many-instance-attributes,
+	too-few-public-methods,
+	too-many-public-methods,
+	too-many-return-statements,
+	too-many-branches,
+	too-many-arguments,
+	too-many-locals,
+	duplicate-code,
+        logging-format-interpolation,
+        import-error,
+	unused-argument,
+        unused-variable,
```

Comment:
I'm really not comfortable with almost any of these checks being disabled.  We really shouldn't have unused-variables in our code or really most of these.  The most notable ones I actually want us to correct with this PR are `unused-variable`, `import-error`, `logging-format-interpolation`, `no-self-use`, `redundant-unittest-assert`, `redifined-outer-name`, `unused-argument`, and `duplicate-code`.  For other more stylistic ones like `too-many-locals`, or `invalid-name` we can possibly deal with these in our RC file by changing properties of the checks like allowing longer method names, or more local variables, but we shouldn't be disabling them all together.  I really think you should try and address all the pylint issues as they are with just the default RC file and use local disable statements. The only good changes to the RC file that make sense to me are increasing the allowed names for methods.